tarutil.py

```diff
@@ -25,19 +25,28 @@ def filtered_tarfile_generator(
     filter_func: typing.Callable[[tarfile.TarInfo], bool]=lambda tarinfo: True,
     chunk_size=tarfile.BLOCKSIZE,
     chunk_callback: typing.Callable[[bytes], None]=None,
+    tarinfo_callback: typing.Callable[[tarfile.TarInfo], tarfile.TarInfo] = lambda tarinfo: tarinfo,
+    finalise: bool = True,
 ) -> typing.Generator[bytes, None, None]:
     '''
     returns a generator yielding a tarfile that will by default contain the same members as
-    the passed tarfile(s) (src_tf). If a filter-function is given, any entries (TarInfo objects)
+    the passed tarfile (src_tf). If a filter-function is given, any entries (TarInfo objects)
     for which this function will return a "falsy" value will be removed from the resulting
-    tarfile stream (which is the actual value-add from this function).
+    tarfile stream (which is the actual value-add from this function). Additionally, a callback
+    function can be given that will be passed all TarInfo objects after filtering to perform
+    modifications like renaming.
 
     This function is particularly useful for streaming. Note that `_FilelikeProxy` can be used
     to wrap a generator yielding an (input-) tarfile-stream.
 
     In combination, this can be used to - in a streaming manner - retrieve a tarfile-stream, e.g.
     using a http-request (e.g. with requests), and upload the filtered tarfile-stream (e.g. again
     with a http-request send e.g. with requests).
+
+    Finally, the `finalise` parameter controls whether the end-of-file marker (two 512-byte blocks
```

Comment:
nit: it is not actually an end-of-file marker, as it might be known from ASCII (eof), rather than a marker denoting the end of a tar stream (which needs not be the end of the underlying file(system))

Reply:
That is specified by the part in parenthesis. The general terminology was taken from https://www.gnu.org/software/tar/manual/html_node/Standard.html